### DISCUSSION TITLE
keyball44 TRRSケーブルで接続した際にキーボードが反応しない。

### DISCUSSION THREAD

**Discussion by tai714** [Yowkees/keyball#583] [Q&A]
下記の事象が発生しており、解決策をご確認いただけないでしょうか。

■事象
TRRSケーブルを繋ぐと、TRRSケーブルで繋いだ先のキーボードの入力を受け付けない。
TRRSケーブルを繋がなければそれぞれのキーボードで入力が可能。
症状から左のイヤホンジャック周りが怪しいと思ったが、原因特定までに至らず。
左キーボードについては、セットのイヤホンジャックを誤って表面につけてしまい、外す際に破損してしまったため、下記リンク先のイヤホンジャックをAmazonで購入して取り付けた。対象のイヤホンジャックだとダメだったのではないかという点も懸念している。
TRRSケーブルの交換を交換したが改善しなかった。

■Amazonで購入したイヤホンジャック
https://www.amazon.co.jp/gp/product/B07LF2KWVD/ref=ppx_yo_dt_b_asin_title_o03_s00?ie=UTF8&psc=1

＜左にUSBを挿した場合＞　両方のキーボードが入力を受け付けない
![左に刺した場合](https://github.com/Yowkees/keyball/assets/33588301/736dc4c2-585c-40f1-9049-1fd5b69a582f)

＜右にUSBを挿した場合＞　左のキーボードのみ入力を受け付けない
![右に刺した場合](https://github.com/Yowkees/keyball/assets/33588301/d108719f-f1a5-4d3b-8785-2a3780c05ceb)

＜左キーボードの基盤＞
![左の半だ](https://github.com/Yowkees/keyball/assets/33588301/3bbf33d0-dc9c-41e6-ac92-3e553c695a79)

＜右キーボードの基盤＞
![右の半だ](https://github.com/Yowkees/keyball/assets/33588301/8440edc6-7c7a-4b9d-b429-6203b06eb4ab)

ファームウェア：Keyball44_via_v1.3.2（両方のキーボードに入れている）

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

症状から左右の通信が上手くいっていないようですね、ケーブルの交換は行っていただいて症状は変わらないということですので、TRRSジャックのはんだ付けを疑いましたが問題はなさそうに見えます。
1つ気になったのが見え方の問題かもしれませんが、青い丸で囲んだProMicroのはんだ付けに隣の端子にはんだが触れてショートしている箇所があるように見えますので、隣の端子に触れないように端子にしっかり馴染むようにはんだ付けを行ってみてください。
また、念のためTRRSジャック部分のはんだ付けもやり直してみてください。
![330846528-8440edc6-7c7a-4b9d-b429-6203b06eb4ab](https://github.com/Yowkees/keyball/assets/143698349/b3524b38-1369-4467-b17d-8e5969b7d49c)

**Reply by Ryo624** [reply]
tai714様もおっしゃる通り、破損した際に取り替えたTRRSジャックがキットと同梱されている物とは違うものを取り付けたことが原因となっている場合がありますので、改善されない場合は取り替えてみてもいいかもしれません。
白銀ラボのウェブショップからキットに同梱している物と同じものを購入できますので、購入をご希望の場合はウェブショップのお問い合わせフォームからご連絡ください。

**Reply by tai714** [reply]
ご確認ありがとうございます。はんだ付けのやり直しでは改善しませんでした。
白銀ラボさんに問い合わせさせていただきます。

**Reply by tai714** [reply]
TRRSジャックを正規のものに変更した所、若干良くなりまし未だしように使用には至っていません。
USBをつなげた方は、入力可能です。
USBをつなげていない方はLEDは転倒しますが、OLEDには情報が出力されず、キー入力もできませんでした。

**Reply by Ryo624** [reply]
テスターをお持ちでしたら下記画像のオレンジの点で示した２点にテスターを当ててみてください。
ピーという音が鳴れば導通しているので、ケーブルやはんだ付けには問題はありません。
![330846528-8440edc6-7c7a-4b9d-b429-6203b06eb4ab](https://github.com/Yowkees/keyball/assets/143698349/17bbbe57-d99b-41c9-8566-d84051703010)
![330846388-3bbf33d0-dc9c-41e6-ac92-3e553c695a79](https://github.com/Yowkees/keyball/assets/143698349/4fa219f2-1498-49dd-a4b6-47e3dded4d3c)

**Reply by tai714** [reply]
指定いただいた箇所は導通していました。

**Reply by tai714** [reply]
電子工作の超ド素人のため、だからなんだという話はできないのですが、左右で抵抗値の違いとかあるのかと見てみたところ、ジャックの所に差異を見つけました。何か参考となるでしょうか。

■左のキーボード
![IMG_2331](https://github.com/Yowkees/keyball/assets/33588301/6902a791-6630-4d25-872a-9a30e3aef60f)

■右のキーボード
![IMG_2330](https://github.com/Yowkees/keyball/assets/33588301/4c20ed78-f2a1-4ff0-b7aa-0979394dfcdf)

**Reply by tai714** [reply]
これのためにテスターまで買ってしまいました笑
もはや愛着の塊です。なんとか直したいところ。

**Reply by Ryo624** [reply]
一度試していただきたいのですが、ProMicroを左右で入れ替えると症状はどうなるか確かめていただいてもよろしいでしょうか。

**Reply by tai714** [reply]
ProMicroを左右入れ替えました。
症状は変わらず、下記の通りです。USBの挿入は左右で試して同じ結果となっています。
・USB-typecを指した側のみ、キー入力できる。
・USB-typecを指した側のみ、OLEDに情報が出力される。
・LEDは両方点灯する。

**Reply by tai714** [reply]
追加で変な動きとしては、
たまに右だけLEDが青に点灯します。基本は両方赤なのですが。
![IMG_2340](https://github.com/Yowkees/keyball/assets/33588301/026ad8e6-c4df-4032-96a5-81e3bbd2d638)

**Reply by Ryo624** [reply]
大変申し訳ありませんが改めてProMicroの裏表とTRRSジャックのはんだ付けをやり直してみていただいてもよろしいでしょうか。
はんだの量が多く丸々としている箇所が多くありますので、富士山のように山型で端子に馴染むようなはんだ付けを意識してみてください。

**Reply by tai714** [reply]
ProMicroの裏面ハンダでピンにハンダを流し込んでProMicroが差し込めなくなってしまいました。ハンダ抜く手段はあるでしょうか。ハンダ吸い取り線で吸ってはいるのですがうまくできません。
ProMicroはさせないし、ピンは基盤から取れそうになく中々絶望的です。

**Reply by Ryo624** [reply]
ピンソケットにはんだが入り込んでしまうと中に入り込んだはんだを除去するのは困難です。
はんだ吸い取り線でProMicro裏面のはんだをなるべく少なくして、基板を傷つけないようにピンソケットは１ピンずつ破壊してしまうつもりで取り除くことになると思われます。

**Reply by tai714** [reply]
了解です。教えていただいた方法で取れました。中々基盤はボロボロで心配ですが、ピンソケットをゲットしてもう少し頑張ります。ピンソケットのサイズ等の情報はあるでしょうか。購入したく。


![IMG_2351](https://github.com/Yowkees/keyball/assets/33588301/4d98ca9b-f1ed-4344-9611-0fd4ef79f8fb)

**Reply by Ryo624** [reply]
返信が遅れて申し訳ありません。

下記リンクから同じ型番のものが購入できます。
https://shop.yushakobo.jp/products/3696

**Reply by tai714** [reply]
購入して、繋がりました！ありがとうございます。
ただ、OLEDがつかなくなってしまいました。おそらくOLEDとUSB-CのVCC（図の赤丸）が導通していないためだと思われるのですが、対処法あるでしょうか。赤丸と裏側（基盤）との導通は取れています
OREDについては、反対側のキーボードで点灯することを確認できており、破損はありません。
![S__22814744](https://github.com/Yowkees/keyball/assets/33588301/327761ac-78e6-4c78-a064-9064ed90a10d)

**Reply by Ryo624** [reply]
OLED用のジャンパ、4Pピンソケットのはんだ付けをやり直してみても改善しないでしょうか。
また、ProMicro裏の12Pピンソケットのはんだ付けも上手く馴染んでいない箇所や、隣の端子に触れてしまっているところがあると導通に不具合が出てしまうので、そちらも確認していただけますと幸いです。

**Reply by tai714** [reply]
ジャンパの接触不良が原因でした。
これで完全に完成しました！長い間サポートいただきありがとうございました。

**Reply by Ryo624** [reply]
無事に解決していただきましてありがとうございます。
ぜひ末永くご愛用いただけましたら幸いです！
